### DISCUSSION TITLE
New Vercel OG Image Gen with Hugo

### DISCUSSION THREAD

**Discussion by devxan** [vercel/community#968] [Help] [answered]
Can I generate meta images/social images on a Hugo build with Vercel's new OG Image Generation [library](https://vercel.com/docs/concepts/functions/edge-functions/og-image-generation)? _(Without nextjs)_

**Comment by amyegan** [comment, marked as answer]
I still have a lot to learn about Hugo, so I can't say how to correctly use an npm package in a Hugo build. There is likely a way to use [`@vercel/og`](https://www.npmjs.com/package/@vercel/og) or [`satori`](https://github.com/vercel/satori) with Hugo for OG image generation. Alternatively, you could use this [og-image template](https://github.com/vercel/og-image) to deploy your own image generator service. 

Hopefully someone else has already tried this and will come share their knowledge here. Or you can give it a try and share what you find! ❤️ 

Additional links that may, or may not, be helpful:
- [Using Bootstrap in Hugo with npm](https://markdotto.com/2022/05/23/bootstrap-hugo-npm/)
- [Three reasons why you should use npm with Hugo](https://henkverlinde.com/blog/three-reasons-why-you-should-use-npm-with-hugo)
- [Using Webpack and npm with Hugo](https://medium.com/pragmatic-programmers/using-webpack-and-npm-with-hugo-35068b7765b2)
- [Introducing OG Image Generation: Fast, dynamic social card images at the Edge](https://vercel.com/blog/introducing-vercel-og-image-generation-fast-dynamic-social-card-images)
